rust/agama-lib/src/users/http_client.rs

```diff
@@ -0,0 +1,80 @@
+use super::client::FirstUser;
+use crate::users::model::{RootConfig, RootPatchSettings};
+use crate::{base_http_client::BaseHTTPClient, error::ServiceError};
+
+pub struct UsersHTTPClient {
+    client: BaseHTTPClient,
+}
+
+impl UsersHTTPClient {
+    pub fn new() -> Result<Self, ServiceError> {
+        Ok(Self {
+            client: BaseHTTPClient::new()?,
+        })
+    }
+
+    pub fn new_with_base(client: BaseHTTPClient) -> Result<Self, ServiceError> {
+        Ok(Self { client })
+    }
+
+    /// Returns the settings for first non admin user
+    pub async fn first_user(&self) -> Result<FirstUser, ServiceError> {
+        self.client.get("/users/first").await
+    }
+
+    /// Set the configuration for the first user
+    pub async fn set_first_user(&self, first_user: &FirstUser) -> Result<(), ServiceError> {
+        let result = self.client.put_void("/users/first", first_user).await;
+        if let Err(ServiceError::BackendError(422, ref issues_s)) = result {
```

Comment:
np: perhaps we could use the [http::status::StatusCode](https://docs.rs/http/latest/http/status/struct.StatusCode.html) from the [HTTP crate](https://docs.rs/http/latest/http/).

Reply:
Yup, but then I can't use `if let` like this... that's fine, a `match` will look good too